### ISSUE TITLE
CSP incorrectly skips adding nonce when `unsafe-inline` is set

### ISSUE THREAD

**Issue by csutter** [rollbar/rollbar-gem#1009]
Rollbar's support for Rails's CSP nonce functionality intentionally skips adding a nonce to the script tag if the `script-src` directive includes `unsafe-inline`: https://github.com/rollbar/rollbar-gem/blob/master/lib/rollbar/middleware/js.rb#L186

The rationale for this is described in the relevant spec:
```ruby
# Browser behavior is undefined when unsafe_inline and the nonce are both present.
# The app should never set both, but if they do, our best behavior is to not use the nonce.
```

However, using both a nonce **and** `unsafe-inline` at the same time is perfectly valid (and indeed sensible) behaviour. It allows the app to maintain some level of backwards compatibility with browsers that support CSP1 but not CSP2.

See [CSP `script-src` on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src):
> Specifying nonce makes a modern browser ignore 'unsafe-inline' which could still be set for older browsers without nonce support.

[Google's CSP checker](https://csp-evaluator.withgoogle.com/) will also issue a warning if a nonce is specified without `unsafe-inline`.

I was going to PR this but figured it might throw up a fair few headaches around changing existing behaviour so wanted to start a conversation first! 😃

**Comment by waltjones** [comment]
@csutter Thank you for this. Those comments were mine, and it looks like I misunderstood that part of the spec.

If I'm now understanding correctly, the change in behavior will just make this case start working. I think a PR would be fine.

**Comment by waltjones** [comment]
Released in v3.1.2.
